gnovm/stdlibs/testing/testing.gno

```diff
@@ -363,3 +413,79 @@ func tRunner(t *T, fn testingFunc, verbose bool) {
 
 	fn(t)
 }
+
+func testSetContext(
+	isOrigin bool,
+	originCaller string,
+	originPkgAddress string,
+	currRealmAddr string, currRealmPkgPath string,
+	origSendDenoms []string, origSendAmounts []int64,
+	origSpendDenoms []string, origSpendAmounts []int64,
+	chainID string,
+	height int64,
+	timeUnix int64, timeNano int64,
+)
+
+func testIssueCoins(addr string, denom []string, amt []int64)
+
+func TestSetContext(ctx Context) {
+	originSendDenom, originSendAmount := expandNative(ctx.OriginSend)
+	originSpendDenom, originSpendAmount := expandNative(ctx.OriginSpend)
+
+	var timestamp, timestampNano int64 = 0, 0
+
+	if !ctx.Time.IsZero() {
+		timestamp = ctx.Time.Unix()
+		timestampNano = 0
+	}
+
+	testSetContext(
+		ctx.IsOrigin,
+		ctx.OriginCaller.String(),
+		ctx.OriginPkgAddress.String(),
+		ctx.CurrentRealm.Address().String(), ctx.CurrentRealm.PkgPath(),
+		originSendDenom, originSendAmount,
+		originSpendDenom, originSpendAmount,
+		ctx.ChainID,
+		ctx.Height,
+		timestamp, timestampNano,
+	)
+}
+
+func SetOriginCaller(origCaller std.Address) {
+	TestSetContext(Context{OriginCaller: origCaller})
+}
+
+func SetOriginSend(send std.Coins) {
+	TestSetContext(Context{OriginSend: send})
+}
+
+func SetOriginSpend(spend std.Coins) {
+	TestSetContext(Context{OriginSpend: spend})
+}
+
+func SetOriginPkgAddress(addr std.Address) {
+	TestSetContext(Context{OriginPkgAddress: addr})
+}
+
+func SetHeight(height int64) {
+	TestSetContext(Context{Height: height})
+}
```

Comment:
I'd say make this work from a `getContext()` unexported function, on top of which we set the value (like OriginCaller, OriginSend, etc.) and then use SetContext; in line with my other comment of not checking the zero value in the Context's fields